app/bundles/LeadBundle/Form/Type/UpdateLeadActionType.php

```diff
@@ -36,17 +38,168 @@ public function buildForm (FormBuilderInterface $builder, array $options)
     {
         /** @var \Mautic\LeadBundle\Model\FieldModel $fieldModel */
         $fieldModel = $this->factory->getModel('lead.field');
-        $leadFields = $fieldModel->getFieldList(false, false);
-
-        foreach ($leadFields as $key => $label) {
-            $builder->add($key, 'text', array(
-                'label'      => $label,
-                'label_attr' => array('class' => 'control-label'),
-                'required'   => false,
-                'attr'       => array(
-                    'class'       => 'form-control'
-                )
-            ));
+        $leadFields = $fieldModel->getEntities(
+            array(
+                'force'          => array(
+                    array(
+                        'column' => 'f.isPublished',
+                        'expr'   => 'eq',
+                        'value'  => true
+                    )
+                ),
+                'hydration_mode' => 'HYDRATE_ARRAY'
+            )
+        );
+
+        foreach ($leadFields as $key => $field) {
+            if ($field['isPublished'] === false) continue;
+            $attr        = array('class' => 'form-control');
+            $properties  = $field['properties'];
+            $type        = $field['type'];
+            $alias       = $field['alias'];
+            $value       = isset($options['data'][$alias]) ? $options['data'][$alias] : '';
+            $constraints = array();
+            if ($type == 'number') {
+                if (empty($properties['precision'])) {
+                    $properties['precision'] = null;
+                } //ensure default locale is used
+                else {
+                    $properties['precision'] = (int) $properties['precision'];
+                }
+
+                $builder->add(
+                    $alias,
+                    $type,
+                    array(
+                        'required'      => false,
+                        'label'         => $field['label'],
+                        'label_attr'    => array('class' => 'control-label'),
+                        'attr'          => $attr,
+                        'data'          => $value,
```

Comment:
Had to use this instead to prevent Symfony from throwing an error that and empty string is not numeric. 

```
                        'data'          => (empty($value)) ? null : $value, // prevent error that it's not a number
```